src/server/scripts/Spells/spell_druid.cpp

```diff
@@ -896,6 +960,74 @@ class spell_dru_omen_of_clarity : public AuraScript
     }
 };
 
+// 274902 - Photosynthesis
+class spell_dru_photosynthesis : public AuraScript
+{
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo
+        ({
+            SPELL_DRUID_LIFEBLOOM,
+            SPELL_DRUID_LIFEBLOOM_FINAL_HEAL,
+            SPELL_DRUID_LIFEBLOOM_OVERGROWTH
+        });
+    }
+
+    bool CheckProc(AuraEffect const* aurEff, ProcEventInfo& eventInfo)
+    {
+        return roll_chance_i(aurEff->GetAmount()) &&
+            (eventInfo.GetActionTarget()->HasAura(SPELL_DRUID_LIFEBLOOM, eventInfo.GetActor()->GetGUID()) ||
+             eventInfo.GetActionTarget()->HasAura(SPELL_DRUID_LIFEBLOOM_OVERGROWTH, eventInfo.GetActor()->GetGUID()));
+    }
+
+    void HandleProc(AuraEffect* aurEff, ProcEventInfo& eventInfo)
+    {
+        GetTarget()->CastSpell(eventInfo.GetActionTarget(), SPELL_DRUID_LIFEBLOOM_FINAL_HEAL, CastSpellExtraArgs(aurEff));
+    }
+
+    void Register() override
+    {
+        DoCheckEffectProc += AuraCheckEffectProcFn(spell_dru_photosynthesis::CheckProc, EFFECT_1, SPELL_AURA_DUMMY);
+        OnEffectProc += AuraEffectProcFn(spell_dru_photosynthesis::HandleProc, EFFECT_1, SPELL_AURA_DUMMY);
+    }
+};
+
+// 274906 - Photosynthesis
+// Called by Lifebloom and Lifebloom (Overgrowth)
+class spell_dru_photosynthesis_effect : public AuraScript
+{
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_DRUID_PHOTOSYNTHESIS_EFFECT })
+            && ValidateSpellEffect({ { SPELL_DRUID_PHOTOSYNTHESIS, EFFECT_0 } });
+    }
+
+    void HandleOnApply(AuraEffect const* aurEff, AuraEffectHandleModes /*mode*/)
+    {
+        Unit* target = GetTarget();
+        if (target->GetGUID() != GetCasterGUID())
+            return;
+
+        if (AuraEffect* photosynthesisEffect = target->GetAuraEffect(SPELL_DRUID_PHOTOSYNTHESIS, EFFECT_0))
+            target->CastSpell(target, SPELL_DRUID_PHOTOSYNTHESIS_EFFECT, CastSpellExtraArgs(aurEff).AddSpellBP0(photosynthesisEffect->GetAmount()));
+    }
+
+    void HandleAfterRemove(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+    {
+        Unit* target = GetTarget();
+        if (target->GetGUID() != GetCasterGUID())
+            return;
+
+        target->RemoveAurasDueToSpell(SPELL_DRUID_PHOTOSYNTHESIS_EFFECT);
```

Comment:
so if the spell id of the spell here and the comment of the script spell id is correct, then i have absolutely no clue what you're trying to do in this hook here